aiida/orm/utils.py

```diff
@@ -50,56 +50,93 @@ def WorkflowFactory(module):
     return BaseFactory(module, Workflow, "aiida.workflows")
 
 
-def load_node(node_id=None, pk=None, uuid=None, parent_class=None):
+def load_node(node_id=None, pk=None, uuid=None, parent_class=None, query_with_dashes=True):
```

Comment:
Is this option really useful? Also, can we add a test to query by start-uuid with 8 and, say, 10 and 16 characters, to see if the method is properly working? (I think this should always query with dashes, and this should work for both backends)